### PR TITLE
search for RIs by name, inchi, or inchikey in addition to CAS

### PR DESCRIPTION
**Pull request by Aariq**
This PR mainly addresses #219.  It adds the ability to search for retention indices from NIST using InChI, InChIkey, or compound name.  

Bugs fixed:

- I discovered and fixed a bug where compounds that only had one entry would produce malformed output.

Dependencies added:

- `polite`.  I switched `nist_ri()` to use if for web scraping and I quite like using it, but I'm happy to remove this dependency if you don't like it.
- `rlang`.  I did a few things with `rlang` that I do not know how to do otherwise.
- `tibble`. I switched the output to a tibble rather than a data.frame.

I'm not sure why so much of the README changed.  It looks like maybe it just didn't get re-built with some previous PRs.

PR task list:
- [x] Update NEWS
- [x] Add tests (if appropriate)
- [x] Update documentation with `devtools::document()`
- [x] Update README examples
- [x] Check package passed

### CODE REVIEW

File: DESCRIPTION [hunk 1]
```diff
@@ -32,7 +32,10 @@ Imports:
     stringr,
     methods,
     dplyr,
-    purrr
+    purrr,
+    polite,
+    rlang,
+    tibble
```
Comment (stitam) [inline DESCRIPTION]: A small consistency issue that just emerged: `as_tibble()` can be `tibble::as_tibble()` or `dplyr::as_tibble()`. `dplyr` imports the function from `tibble` so probably they are the same but `tibble` is the original source, so I agree with using `tibble::as_tibble()`. In PR #230 I used `dplyr::as_tibble()` but let's keep this consistent and your approach is better. I'll add this to the contributing guide as well.

**Comment by Aariq** [comment]
You know, I just realized I changed the scrape delay, but never tested this on large numbers of queries.  Let me do some testing and see if I don't get my IP banned before merging this PR

**Comment by Aariq** [comment]
Ok, fixed it up so it works with NAs better. New tests are all passing locally.

File: R/nist.R [hunk 2]
```diff
@@ -231,18 +287,46 @@ tidy_ritable <- function(ri_xml) {
...
+        polarity = polarity,
+        temp_prog = temp_prog
+      )
+    )
+
+  querynames <- query
+  querynames [is.na(querynames)] <- ".NA"
+  ri_xmls <- setNames(ri_xmls, querynames)
+
+  ri_tables <- map_dfr(ri_xmls, tidy_ritable, .id = "query") %>%
```
Comment (stitam) [inline R/nist.R]: `purrr::map_dfr()`?

File: R/nist.R [hunk 3]
```diff
@@ -231,18 +287,46 @@ tidy_ritable <- function(ri_xml) {
...
+    mutate(query = na_if(query, ".NA"))
```
Comment (stitam) [inline R/nist.R]: `dplyr::mutate()`?

File: tests/testthat/test-nist.R [hunk 4]
```diff
@@ -1,19 +1,125 @@
 context("nist")
 
-a <- nist_ri("78-70-6")
-b <- nist_ri("78-70-6", type = 'linear', polarity = 'non-polar')
-c <- nist_ri("78-70-6", type = 'linear', temp_prog = 'custom')
+test_that("nist_ri() warns when no results", {
+  skip_on_cran()
+  expect_warning(nist_ri("78-70-6", from = "cas", type = 'linear', polarity = 'non-polar'),
```
Comment (stitam) [inline tests/testthat/test-nist.R]: Can you check the file with `lintr::lint()` for conformance to the style guide? This line is a bit too long.

File: NEWS [hunk 5]
```diff
@@ -3,10 +3,13 @@ webchem 0.6.0
 
 NEW FEATURES
 
+* nist_ri() now can search by name, InChI, InChIkey, or CAS#.  The `cas` argument is deprecated.  Use `query` instead with `from = "cas"`
```
[the remark this answers is not in the document]
Comment (Aariq) [reply NEWS]: I changed it to version 0.5.0.9000 which is what `usethis::use_dev_version()` and http://r-pkgs.had.co.nz/description.html#version suggest.

File: R/nist.R [hunk 6]
```diff
@@ -1,67 +1,116 @@
...
+      "name" = "Name",
+      "inchi" = "InChI",
+      "inchikey" = "InChI",
+      "cas" = "ID"
+    ))
+
+    # Open session
+    #TODO: get rid of "no encoding supplied: defaulting to UTF-9" message
+    #properly
+    session <- suppressMessages(bow("https://webbook.nist.gov/cgi/cbook.cgi"))
```
[the remark this answers is not in the document]
Comment (Aariq) [reply R/nist.R]: fixed

File: R/nist.R [hunk 7]
```diff
@@ -1,67 +1,116 @@
...
+    set_scrape_delay(rgamma(1, shape = 15, scale = 1/10))
```
[the remark this answers is not in the document]
Comment (Aariq) [reply R/nist.R]: fixed

File: R/nist.R [hunk 8]
```diff
@@ -1,67 +1,116 @@
...
+      return(NA)
     } else {
-      ri_xml <- ri_xml.all
-    }
 
+      if (from == "cas") {
+        ID <- paste0("C", gsub("-", "", query))
+      } else {
+        page <-
+          scrape(session,
```
[the remark this answers is not in the document]
Comment (Aariq) [reply R/nist.R]: fixed

File: R/nist.R [hunk 9]
```diff
@@ -1,67 +1,116 @@
...
+          return(NA)
+        } else {
+          ID <- str_extract(gaschrom, "(?<=ID=).+?(?=&)")
+        }
+      }
+      #scrape RI table
+      type_str <-
+        toupper(paste(type, "RI", polarity, temp_prog, sep = "-"))
+
+      page <- scrape(session,
```
[the remark this answers is not in the document]
Comment (Aariq) [reply R/nist.R]: fixed

File: R/nist.R [hunk 10]
```diff
@@ -151,62 +202,67 @@ tidy_ritable <- function(ri_xml) {
...
+      dplyr::select("RI", "type", "phase", everything())
   }
   return(output)
 }
 
 
 #' Retrieve retention indices from NIST
 #' @description This function scrapes NIST for literature retention indices
-#'  given CAS numbers as an input.
+#'   given CAS numbers as an input.
```
[the remark this answers is not in the document]
Comment (stitam) [reply R/nist.R:245]: This function scrapes NIST for literature retention indices.

**Comment by stitam** [comment]
I ran R CMD check locally and it succeeded.

**Pull request merged**